ide/projectui/src/org/netbeans/modules/project/ui/groups/Group.java

```diff
@@ -562,7 +562,15 @@ static void open(final Group g, String oldGroupName, boolean isNewGroup, Prefere
             h.start(200);
             ProjectUtilities.WaitCursor.show();
             final OpenProjectList opl = OpenProjectList.getDefault();
-            Set<Project> oldOpen = new HashSet<Project>(Arrays.asList(opl.getOpenProjects()));
+            
+            Set<Project> oldOpen = new HashSet<>();
+            for (Project open : Arrays.asList(opl.getOpenProjects())) {
+                // TODO fix this properly
+                // unbox potential fod.FeatureNonProject wrapper since it breaks Sets/Maps due to incompatible hashcode impls
+                Project real = open.getLookup().lookup(Project.class);
+                oldOpen.add(real != null ? real : open);
+            }
+
```

Comment:
this is the workaround

Reply:
The TODO comment would be improved by suggesting what the proper fix might look like.

What exactly is a FeatureNonProject? I found the class but it's light on comments. Some placeholder object for projects that are not fully loaded yet?

Reply:
it appears to be a placeholder which is used by the ergonomics mechanism to represent project functionality before the cluster for the functionality is loaded. At this point, the cluster is already loaded, but code has still the activation proxy stored.

Reply:
> The TODO comment would be improved by suggesting what the proper fix might look like.

IMO: the proper fix would be to repair equals/hashcode. Once done `opl.getOpenProjects()` should be changed to return only unboxed projects - since I don't think projects can be open without their cluster loaded.

However: realistically, I am not sure if this can be done at this point since it is quite risky. The getter is exposed via public API and fixing object identity mechanics could break other things, given how many workaround already exist (e.g the `a.equals(b) || b.equals(a)` situation). If at all this must be done carefully and might need some refactoring first before attempting it.